crates/cranelift/src/compiler.rs

```diff
@@ -332,8 +332,9 @@ impl wasmtime_environ::Compiler for Compiler {
             let memory_offset = if ofs.num_imported_memories > 0 {
                 ModuleMemoryOffset::Imported(ofs.vmctx_vmmemory_import(MemoryIndex::new(0)))
             } else if ofs.num_defined_memories > 0 {
+                // TODO shared?
```

Comment:
It is unclear what to do at this point if the module's memory is a shared one... The current logic below is incorrect.

Reply:
Given that this is only debuginfo related and debuginfo is off-by-default I think it'd be fine to just return an error here for shared memories and have someone else come along later and figure this out if necessary.